File: bachngocsach/src/tab.lua
local url, page = ...
if text:is_empty(page) then
    page = 0
end
local doc = http:get(url .. "?page=" .. page):html()
if doc ~= nil then
    local el = doc:select("ul.content-grid > li > div")
    local list = {}
    for i = 0, el:size() - 1 do
        local e = el:get(i)
        local item = {}
        item["name"] = e:select("div.recent-truyen a"):text()
        item["link"] = e:select("div.recent-truyen a"):attr("href")
        item["cover"] = e:select("div.recent-anhbia img"):attr("src")
        item["description"] = e:select("div.recent-chuong a"):text()
        item["host"] = "https://bachngocsach.com"
        table.insert(list, item)
    end

    local pageNext = regexp:find_last(doc:select("li.pager-next > a"):attr("href"), "page=(\\d+)")
    return response:success(list, pageNext)
end
return nil

File: bachngocsach/src/tab.js
load('config.js');
function execute(url, page) {
    if (!page) page = '0';

    let response = fetch(url + "/?page=" + page);
    if (response.ok) {
        let doc = response.html();
        if (doc.select("title").text().includes("Đăng nhập để đọc truyện")) {
            return Response.error("Bạn cần đăng nhập hoặc tạo tài khoản mới để tiếp tục đọc truyện.");
        }
        let next = doc.select(".pager-next").last().select("a").attr("href").match(/page=(\d+)/);
        if (next) next = next[1];

        let novelList = [];
        doc.select("ul.content-grid > li > div").forEach(e => novelList.push({
            name: e.select("div.recent-truyen a").text(),
            link: e.select("div.recent-truyen a").attr("href"),
            cover: e.select("div.recent-anhbia img").attr("src"),
            description: e.select("div.recent-chuong a").text(),
            host: BASE_URL
        }));
        return Response.success(novelList, next);
    }
    return null;
}